### PR TITLE
Add rhel8/rhel7/centos7/fedora version for nginx 1.14

### PR DESCRIPTION
**Pull request by pkubatrh**

### CODE REVIEW

File: 1.14/Dockerfile.rhel8 [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    mkdir -p ${NGINX_APP_ROOT}/etc/nginx.default.d/ && \
+    mkdir -p ${NGINX_APP_ROOT}/src/nginx-start/ && \
+    mkdir -p ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    chmod -R a+rwx ${NGINX_APP_ROOT}/etc && \
+    chmod -R a+rwx /var/lib/nginx && \
+    chmod -R a+rwx ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    chown -R 1001:0 ${NGINX_APP_ROOT} && \
+    chown -R 1001:0 /var/lib/nginx && \
+    chown -R 1001:0 ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    # FIXME: Not sure if this is safe to do, just a hack to make the image work
```
Comment (omron93) [inline 1.14/Dockerfile.rhel8]: Only `chmod a+rwx /run` works for me too. IMHO above chmod/chown section could be reduced too, but someone more experienced with nginx would be better.

Comment (pkubatrh) [reply 1.14/Dockerfile.rhel8]: @uhliarik PTAL if you find some time ^

File: 1.14/Dockerfile.rhel8 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    chmod -R a+rwx /var/run && \
+    chown -R 1001:0 /var/run && \
+    rpm-file-permissions
+
+USER 1001
+
+# Not using VOLUME statement since it's not working in OpenShift Online:
+# https://github.com/sclorg/httpd-container/issues/30
+# VOLUME ["/opt/rh/rh-nginx114/root/usr/share/nginx/html"]
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply 1.14/Dockerfile.rhel8]: Will change to the volumes from Fedora image, thanks.

File: 1.14/Dockerfile.rhel8 [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+# Copy extra files to the image.
+COPY ./root/ /
+
+# In order to drop the root user, we have to make some directories world
+# writeable as OpenShift default security model is to run the container under
+# random UID.
+RUN sed -i -f ${NGINX_APP_ROOT}/nginxconf-fed.sed ${NGINX_CONF_PATH} && \
+    chmod a+rwx ${NGINX_CONF_PATH} && \
+    mkdir -p ${NGINX_APP_ROOT}/etc/nginx.d/ && \
```
[the remark this answers is not in the document]
Comment (pkubatrh) [reply 1.14/Dockerfile.rhel8]: Thanks for spotting this, will add in next commit.

**Comment by pkubatrh** [comment]
> ./content_sets.yml is not required for 1.14/

Not required at all in upstream github. We will have differing `content_sets.yml` files for RHEL7 and RHEL8 so did not add any version right now.

> 8/./root/opt/app-root/etc/scl_enable is going to fail in rhel8 image (see https://github.com/sclorg/s2i-base-container/pull/177/files#r239427163). Same probles is going to be with nodejs-10 RHEL7 and RHEL8 variant.

Will need to be fixed in s2i-core first. That said it does not actually fail afaics, it only warns that no SCLs could be found for enabling which is sliglty less bad.

> Would you mind to add 1.14/Dockerfile.fedora and 1.14/Dockerfile.rhel7 into this PR? Changes from version 1.12/ are really small.

Will do.

File: 1.14/Dockerfile.rhel8 (hunk 3 above)
[the remark this answers is not in the document]
Comment (pkubatrh) [reply 1.14/Dockerfile.rhel8]: Is there a reason for not having the same PR applied to rhel7 images?

**Comment by pkubatrh** [comment]
Hopefully all changes should be in now.

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by pkubatrh** [comment]
[test-openshift]

**Comment by patrick-tavares** [comment]
Would love to have this PR merged! I have a pending project in which I would like to use Nginx 1.14 containers. Granted most of us just returned from the holidays but, any update?

**Comment by pkubatrh** [comment]
@uhliarik @hhorak PTAL

**Comment by hhorak** [comment]
LGTM, merging

**Pull request merged**